crates/sparrow-interfaces/src/source.rs

```diff
@@ -0,0 +1,12 @@
+use arrow_schema::SchemaRef;
+use futures::stream::BoxStream;
+use sparrow_batch::Batch;
+use sparrow_core::ReadConfig;
+
+use crate::Error;
+
+/// Defines how a source provides data to the execution layer.
+pub trait Source: Send + Sync {
+    fn schema(&self) -> SchemaRef;
```

Comment:
`projected` or `prepared`?

Reply:
I suspect this is actually just the `raw` schema - when a user creates a source, all we can do is get the schema of the given input. So..I guess it could be the `prepared` schema too. I don't know how this method will be used yet though. The `_ffi.parquet_schema` method currently just gets the schema from the given file. 

`Execution` also has a `schema`, which is used to construct a `table` from batches to go to pandas. So that's the result schema.

Reply:
Pick one? and rename method to match for now